tools/generate_tables_sql_and_testNames.php

```diff
@@ -1,118 +1,224 @@
+ *
+ * Ex cmd without SQL execution:
+ * php generate_tables_sql_and_testNames.php <
+ * ../instruments/[instrument_name].linst false
+ *
+ * @category Tools
+ * @package  Behavioural
+ * @author   Loris Team <loris.mni@bic.mni.mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris
  */
-
-//Ensure php version compatability
-//taken from php.net notes
-
-// require all relevant OO class libraries
-require_once __DIR__ . "/../vendor/autoload.php";
-require_once "../php/libraries/Database.class.inc";
-require_once "../php/libraries/NDB_Config.class.inc";
-require_once "../php/libraries/NDB_BVL_Instrument.class.inc";
-
-$data = stream_get_contents(STDIN);
-
-$instruments=explode("{-@-}",trim($data));
-
-$tblCount=0;
-$parameterCount=0;
-$pages = array();
-foreach($instruments AS $instrument){
-    $catId="";
-    $items=explode("\n",trim($instrument));
-    foreach($items AS $item){
-        $paramId="";
-        $bits=explode("{@}",trim($item));
-        if(preg_match("/Examiner[0-9]*/" , $bits[1])){
-            continue;
-        }
-        switch($bits[0]){
-            //generate the CREATE TABLE syntax
+// require all relevant libraries
+require_once "./generic_includes.php";
+$execute_mysql_query = ($argv[2] ?? true) === true;
+$data          = file_get_contents($argv[1]);
+$instruments   = explode("{-@-}", trim($data));
+$error_message = "";
+// Do validation
+if (!is_array($instruments)) {
+    $error_message = "Instrument file could not be processed.";
+} else {
+    $pages = array();
+    $sql_query_statement  = "";
+    $sql_query_table_name = "";
+    $sql_query_columns    = array();
+    // Setup Database object.
+    $factory   = \NDB_Factory::singleton();
+    $db        = $factory->database();
+    foreach ($instruments as $instrument) {
+        $items = explode("\n", trim($instrument));
+        foreach ($items as $item) {
+            $bits = explode("{@}", trim($item));
+            if (preg_match("/Examiner[0-9]*/", $bits[1])) {
+                continue;
+            }
+            switch($bits[0]) {
+                // Generate the CREATE TABLE syntax
             case "table":
-                $tablename = $bits[1];
-
-                $filename="../project/tables_sql/".$bits[1].".sql";
-                $output="CREATE TABLE `$bits[1]` (\n";
-                $output.="`CommentID` varchar(255) NOT NULL default '',
-                          `UserID` varchar(255) default NULL,
-                          `Examiner` varchar(255) default NULL,
-                          `Testdate` timestamp DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
-                          `Data_entry_completion_status` enum('Incomplete','Complete') NOT NULL default 'Incomplete',\n";
-            break;
+                $sql_query_table_name = str_replace('`', '', $db->escape($bits[1]));
+                $file_name = "../project/tables_sql/".$bits[1].".sql";
```

Comment:
```suggestion
                $file_name = "../project/tables_sql/".$sql_query_table_name.".sql";
```
If backtick are present in $bits[1], this might pose problem with persons that are not bash expert.